src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -235,6 +236,51 @@ struct aie_shim_info : aie_metadata
   }
 };
 
+struct kds_cu_stat
+{
+  using result_type = query::kds_cu_stat::result_type;
+  using data_type = query::kds_cu_stat::data_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type)
+  {
+    auto edev = get_edgedev(device);
+
+    using tokenizer = boost::tokenizer< boost::char_separator<char> >;
+    std::vector<std::string> stats;
+    std::string errmsg;
+
+    // The kds_custat_raw is printing in formatted string of each line
+    // Format: "%d,%s:%s,0x%lx,0x%x,%lu"
+    // Using comma as separator.
+    edev->sysfs_get("kds_custat_raw", errmsg, stats);
+    if (!errmsg.empty())
+      throw std::runtime_error(errmsg);
+
+    result_type cuStats;
+    for (auto& line : stats) {
+      boost::char_separator<char> sep(",");
+      tokenizer tokens(line, sep);
+
+      if (std::distance(tokens.begin(), tokens.end()) != 5)
+        throw std::runtime_error("CU statistic sysfs node corrupted");
+
+      data_type data;
+      const int radix = 16;
+      tokenizer::iterator tok_it = tokens.begin();
+      data.index     = std::stoi(std::string(*tok_it++));
+      data.name      = std::string(*tok_it++);
+      data.base_addr = std::stoull(std::string(*tok_it++), nullptr, radix);
+      data.status    = std::stoul(std::string(*tok_it++), nullptr, radix);
+      data.usages    = std::stoul(std::string(*tok_it++));
+
+      cuStats.push_back(data);
```

Comment:
push_back(std::move(data))
or emplace_back construction of data.  Personally prefer push_back in this case.